autorest/codegen/serializers/builder_serializer.py

```diff
@@ -837,6 +839,15 @@ def _call_request_builder_helper(
                 parameter.serialized_name not in builder.body_kwargs_to_pass_to_request_builder
             ):
                 continue
+            if (
+                is_next_request and
+                not bool(builder.next_request_builder) and
+                self.code_model.options["version_tolerant"] and
```

Comment:
https://github.com/Azure/autorest.python/issues/1165 is found in `azure-mgmt-recoveryservicesbackup`, so I think we shall remove the limitation `version_tolerant`

Reply:
Yeah, we ended up patching legacy code. The issue is it's a breaking change to change how we deal with next link calls, so we can't enable this for legacy, even if it might help one team (because other teams might be broken by it). But yeah, definitely good point because there is a legacy team affected by it

Reply:
If we are afraid to break current legacy code, how about adding flag like `--disable-filter-query-params` to permit some service like `recoveryservicesbackup` to avoid duplicated filter params in legacy code.

Reply:
added flag `--reformat-next-link`, which defaults to `True` for legacy code generations, is forced to False for DPG generations. Thanks!